### PR TITLE
contrib/libtoxcore: new package (0.2.19)

### PR DESCRIPTION
**Pull request by Ogromny**

### CODE REVIEW

File: contrib/libtoxcore/template.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+url = "https://github.com/TokTok/c-toxcore"
+source = f"{url}/releases/download/v{pkgver}/c-toxcore-{pkgver}.tar.gz"
+sha256 = "8b418f6470db085cf59a9915685613556556df2bf427148f1814b7b118628594"
+
+
+def post_install(self):
+    self.install_license("LICENSE")
+
+
+@subpackage(f"{pkgname}-devel")
```
Comment (triallax) [inline contrib/libtoxcore/template.py:29]:
```suggestion
@subpackage(f"libtoxcore-devel")
```

**Comment by nekopsykose** [comment]
call it c-toxcore

**Comment by nekopsykose** [comment]
the tests also have to be enabled

File: contrib/libtoxcore/template.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "pkgconf",
+]
+makedepends = [
+    "libconfig-devel",
+    "libsodium-devel",
+    "libvpx-devel",
+    "linux-headers",
+    "opus-devel",
+]
+pkgdesc = "Future of online communications"
```
Comment (q66) [inline contrib/libtoxcore/template.py:17]: come up with a better pkgdesc

**Review by triallax — CHANGES_REQUESTED** [review]

**Comment by q66** [comment]
also not much interested in standalone libraries that cannot be tested with anything

if you have an application then do that together

File: contrib/libtoxcore/template.py (hunk 2 above)
Comment (Ogromny) [reply contrib/libtoxcore/template.py:17]: Is "Tox communication project" good ?

**Comment by Ogromny** [comment]
> also not much interested in standalone libraries that cannot be tested with anything
> 
> if you have an application then do that together

I was planning to add `toxic` (https://github.com/tokTok/toxic) after this commit, should I add it in this commit ?

**Comment by nekopsykose** [comment]
separate commits, in this merge request

**Comment by Ogromny** [comment]
> separate commits, in this merge request

Ok I will add it asap !

**Comment by nekopsykose** [comment]
you need separate commits for every package

File: contrib/toxic/patches/makefile.patch [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ 
+ LIBS = toxcore ncursesw libconfig libcurl
+ 
+-CFLAGS ?= -std=c11 -pthread -Wall -Wpedantic -Wunused -fstack-protector-all -Wvla -Wno-missing-braces -Wmissing-prototypes -Wcast-align -Wcast-qual -Wmissing-declarations -Wshadow -Wunused-macros -Wformat-nonliteral -Wno-unknown-pragmas
++CFLAGS = -std=gnu11 -pthread -Wall -Wpedantic -Wunused -fstack-protector-all -Wvla -Wno-missing-braces -Wmissing-prototypes -Wcast-align -Wcast-qual -Wmissing-declarations -Wshadow -Wunused-macros -Wformat-nonliteral -Wno-unknown-pragmas
+ CFLAGS += '-DTOXICVER="$(VERSION)"' -DHAVE_WIDECHAR -D_XOPEN_SOURCE_EXTENDED -D_FILE_OFFSET_BITS=64
+ CFLAGS += '-DPACKAGE_DATADIR="$(abspath $(DATADIR))"'
+ CFLAGS += ${USER_CFLAGS}
+-LDFLAGS ?=
++LDFLAGS =
```
Comment (nekopsykose) [inline contrib/toxic/patches/makefile.patch:13]: this doesn't do anything whatsoever as it's already correct

File: contrib/toxic/patches/makefile.patch [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+-	@$(CC) $(CFLAGS) -o $(BUILD_DIR)/osx_video.o -c $(SRC_DIR)/$(OSX_VIDEO)
++	$(CC) $(CFLAGS) -o $(BUILD_DIR)/osx_video.o -c $(SRC_DIR)/$(OSX_VIDEO)
+ 
+ $(BUILD_DIR)/%.o: $(SRC_DIR)/%.c
+ 	@if [ ! -e $(BUILD_DIR) ]; then \
+ 		mkdir -p $(BUILD_DIR) ;\
+ 	fi
+-	@echo "  CC    $(@:$(BUILD_DIR)/%=%)"
+-	@$(CC) $(CFLAGS) -o $(BUILD_DIR)/$*.o -c $(SRC_DIR)/$*.c
++	$(CC) $(CFLAGS) -o $(BUILD_DIR)/$*.o -c $(SRC_DIR)/$*.c
```
Comment (nekopsykose) [inline contrib/toxic/patches/makefile.patch:36]: these also do nothing at all, what is this patch meant to do? it doesn't fix anything

File: contrib/toxic/template.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+pkgname = "toxic"
+pkgver = "0.15.1"
+pkgrel = 0
+build_style = "makefile"
+make_cmd = "gmake"
```
Comment (nekopsykose) [inline contrib/toxic/template.py:5]: you need make_use_env=True as well

File: contrib/toxic/template.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    "c-toxcore-devel",
+    "freealut-devel",
+    "libconfig-devel",
+    "libcurl-devel",
+    "libnotify-devel",
+    "libpng-devel",
+    "libx11-devel",
+    "linux-headers",
+    "ncurses-devel",
+    "openal-soft",
```
Comment (nekopsykose) [inline contrib/toxic/template.py:25]:
```suggestion
```

File: contrib/toxic/patches/makefile.patch (hunk 3 above)
Comment (Ogromny) [reply contrib/toxic/patches/makefile.patch]: My bad I had issue before adding `make_use_env` and so I tried the alpine's patches, I forgot to remove it

File: contrib/freealut/template.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+pkgver = "1.1.0"
+pkgrel = 0
+build_style = "cmake"
+hostmakedepends = ["cmake", "ninja", "pkgconf"]
+makedepends = ["openal-soft-devel"]
+pkgdesc = "OpenAL Utility Toolkit (ALUT)"
+maintainer = "ogromny <ogromnycoding@gmail.com>"
+license = "GPL-2.0-or-later"
+url = "https://github.com/vancegroup/freealut"
+source = f"{url}/archive/refs/tags/freealut_1_1_0.tar.gz"
```
Comment (nekopsykose) [inline contrib/freealut/template.py:11]:
```suggestion
source = f"{url}/archive/refs/tags/freealut_{pkgver.replace('.', '_')}.tar.gz"
```

File: contrib/c-toxcore/template.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    "opus-devel",
+]
+checkdepends = ["gtest-devel"]
+pkgdesc = "Tox communication project"
+maintainer = "ogromny <ogromnycoding@gmail.com>"
+license = "GPL-3.0-only"
+url = "https://github.com/TokTok/c-toxcore"
+source = f"{url}/releases/download/v{pkgver}/c-toxcore-{pkgver}.tar.gz"
+sha256 = "8b418f6470db085cf59a9915685613556556df2bf427148f1814b7b118628594"
+options = ["check"]
```
Comment (nekopsykose) [inline contrib/c-toxcore/template.py:25]:
```suggestion
```

default

File: contrib/freealut/template.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+build_style = "cmake"
+hostmakedepends = ["cmake", "ninja", "pkgconf"]
+makedepends = ["openal-soft-devel"]
+pkgdesc = "OpenAL Utility Toolkit (ALUT)"
+maintainer = "ogromny <ogromnycoding@gmail.com>"
+license = "GPL-2.0-or-later"
+url = "https://github.com/vancegroup/freealut"
+source = f"{url}/archive/refs/tags/freealut_{pkgver.replace('.', '_')}.tar.gz"
+sha256 = "c3880d7dbb90c7db54e1d6b88aa0a34dd9e2d828fd389be0d2cbb2632b0885dd"
+# freealut has no tests
```
Comment (nekopsykose) [inline contrib/freealut/template.py:13]: there are, they're just manually ran from the build dir. same for toxic below apparently

you could see if they make sense to run

File: contrib/freealut/template.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+pkgname = "freealut"
+pkgver = "1.1.0"
+pkgrel = 0
+build_style = "cmake"
+hostmakedepends = ["cmake", "ninja", "pkgconf"]
+makedepends = ["openal-soft-devel"]
+pkgdesc = "OpenAL Utility Toolkit (ALUT)"
```
Comment (nekopsykose) [inline contrib/freealut/template.py:7]:
```suggestion
pkgdesc = "OpenAL utility toolkit"
```

File: contrib/freealut/template.py (hunk 7 above)
Comment (Ogromny) [reply contrib/freealut/template.py]: Thanks ! It should be good now

File: contrib/freealut/template.py (hunk 9 above)
Comment (Ogromny) [reply contrib/freealut/template.py]: I don't see the point of running the test of freealut, `toxic`'s tests are using docker, and if toxic compiles it should just run fine, so I don't think the tests are mandatory for these two, what do you think ?

Comment (nekopsykose) [reply contrib/freealut/template.py]: sure

**Pull request merged**